s3/src/main/scala/blobstore/s3/S3Store.scala

```diff
@@ -24,16 +24,19 @@ import cats.syntax.flatMap._
 import fs2.{Chunk, Sink, Stream}
 
 import scala.collection.JavaConverters._
-import com.amazonaws.services.s3.{AmazonS3, AmazonS3ClientBuilder}
+import com.amazonaws.services.s3.AmazonS3
 import com.amazonaws.services.s3.model.{CopyObjectRequest, ListObjectsRequest, ObjectListing, ObjectMetadata}
+import com.amazonaws.services.s3.transfer.{TransferManager, TransferManagerBuilder}
 
 import scala.concurrent.ExecutionContext
 
-case class S3Store[F[_] : ConcurrentEffect : ContextShift](s3: AmazonS3, sse: Option[String] = None, blockingExecutionContext: ExecutionContext)(implicit F: Effect[F]) extends Store[F] {
+case class S3Store[F[_] : ConcurrentEffect : ContextShift](tm: TransferManager, sse: Option[String] = None, blockingExecutionContext: ExecutionContext)(implicit F: Effect[F]) extends Store[F] {
+
+  final val s3: AmazonS3 = tm.getAmazonS3Client
```

Comment:
does this need to be `final val` or just `val`? A `val` cannot change once assigned. Only difference with `final val` is that it can't be overridden in a subclass or trait

I suggest removing `final`. If someone wanted to extend this class in their own code, this would make it easier to override the s3 client

Reply:
one should never extend a case class though (in fact I think that all case classes should be declared final)

Reply:
I believe you but why is that? (not that I can think of a good reason _to_ do it)

Reply:
Apologies in advance for getting involved in a foreign conversation...
But, [here](https://stackoverflow.com/questions/34561614/should-i-use-the-final-modifier-when-declaring-case-classes) you can find a good explanation of why a `case class` should always be **final**.
BTW, [**Scapegoat**](https://github.com/sksamuel/scapegoat) now also warns you if you don't mark them final _(FinalModifierOnCaseClass)_.